lib/deploy/events/apiGateway/iamRole.js

```diff
@@ -14,15 +13,58 @@ module.exports = {
       return BbPromise.resolve();
     }
 
-    const iamRoleApiGatewayToStepFunctionsTemplate =
-      JSON.stringify(this.serverless.utils.readFileSync(
-      path.join(__dirname,
-        'apigateway-to-stepfunctions-assume-role.json'))
-    );
+    const iamRoleApiGatewayToStepFunctionsAction = [
+      'states:StartExecution',
+    ];
+
+    // generate IAM Role action by http.action parameter.
+    this.pluginhttpValidated.events.forEach((event) => {
+      if (_.has(event, 'http.action')) {
+        const actionName = `states:${event.http.action}`;
+
+        if (iamRoleApiGatewayToStepFunctionsAction.indexOf(actionName) === -1) {
+          iamRoleApiGatewayToStepFunctionsAction.push(actionName);
+        }
+      }
+    });
+
+    const iamRoleApiGatewayToStepFunctions = {
+      Type: 'AWS::IAM::Role',
+      Properties: {
+        AssumeRolePolicyDocument: {
+          Version: '2012-10-17',
+          Statement: [
+            {
+              Effect: 'Allow',
+              Principal: {
+                Service: 'apigateway.amazonaws.com',
+              },
+              Action: 'sts:AssumeRole',
+            },
+          ],
+        },
+        Policies: [
+          {
+            PolicyName: 'apigatewaytostepfunctions',
+            PolicyDocument: {
+              Version: '2012-10-17',
+              Statement: [
+                {
+                  Effect: 'Allow',
+                  Action: iamRoleApiGatewayToStepFunctionsAction,
+                  Resource: '*',
```

Comment:
A minor thing, we can change this later in another PR, but we should be able to restrict the resource to just the state machine in question.